### DISCUSSION TITLE
Combine C and C++ files

### DISCUSSION THREAD

**Discussion by andreamancuso** [emscripten-core/emscripten#22137] [General]
Hi,

Does anyone have examples of CMakeLists.txt files referencing both C and C++ files? I'm trying to add https://github.com/maplibre/maplibre-native as a dependency to my project and I got stuck with https://bitbucket.org/alekseyt/nunicode which is a C99 compliant library.

Specifically, I am getting: `error: invalid argument '-std=c++23' not allowed with 'C'`

Here's the code: https://github.com/andreamancuso/react-wasm/pull/37/files#diff-819e708b98ec981f4fc20bf8190a71977859972c788459cf8b408f998f944aaa

Any help/guidance with this would be much appreciated!

**Comment by sbc100** [comment]
The problem is the line `add_definitions(-std=c++23)`.  

The correct way to set the C++ standard is to use `CXX_STANDARD`: https://cmake.org/cmake/help/latest/prop_tgt/CXX_STANDARD.html.    If you don't want to use that you could use `CMAKE_CXX_FLAGS`.   Both of these methods mean that the flags won't get passed when building plain C files.
